### PR TITLE
Started adding logs to application

### LINKED ISSUE #311: Add logging.
We should log every action taken on the platform.

### PR DESCRIPTION
**Pull request by Naapperas**
Closes #311

### CODE REVIEW

File: app/Actions/Fortify/CreateNewUser.php [hunk 1]
```diff
@@ -27,13 +29,17 @@ public function create(array $input): User
...
+        $data = [
             'name' => $input['name'],
             'email' => $input['email'],
             'password' => Hash::make($input['password']),
             'usertype_id' => '0',
             'usertype_type' => Participant::class,
-        ]);
+        ];
+
+        Log::info('Creating user with input: {input}', ['input' => Json::encode($data, true)]);
```
Comment (ttoino) [inline app/Actions/Fortify/CreateNewUser.php]: I don't think logging a user's password is a great idea 🥴

File: app/Http/Controllers/EventController.php [hunk 2]
```diff
@@ -52,26 +53,45 @@ public function show(Request $request, Event $event)
...
 
         if ($edition === null) {
             return response('No edition found', 500);
         }
 
+        /** @var User|null */
+        $user = $request->user();
+
+        if ($user === null) {
+            return redirect()->back()->dangerBanner('Tens que ter sessão iniciada para fazer esta ação');
```
Comment (ttoino) [inline app/Http/Controllers/EventController.php]: I don't like this error message, it sounds too robotic

File: app/Http/Controllers/QuestController.php [hunk 3]
```diff
@@ -23,11 +24,29 @@ public function give(Request $request, Quest $quest)
...
         }
 
-        if (Gate::denies('give', [$quest, $enrollment])) {
+        /** @var User|null */
+        $user = $request->user();
+
+        if ($user === null) {
+            Log::warning('Unauthenticated user attempted to give quest to participant');
+
+            return redirect()->back()->dangerBanner('Tens que estar autenticado para efetuar esta ação');
```
Comment (ttoino) [inline app/Http/Controllers/QuestController.php]: Same as above

**Review by ttoino — CHANGES_REQUESTED** [review]

**Comment by ttoino** [comment]
Actually, now that I think about it, are those error messages when the user is not logged in even required? The routs have the fortify middleware anyway

File: app/Actions/Fortify/CreateNewUser.php (hunk 1 above)
Comment (Naapperas) [reply app/Actions/Fortify/CreateNewUser.php]: :sob: my bad

File: app/Http/Controllers/EventController.php (hunk 2 above)
Comment (Naapperas) [reply app/Http/Controllers/EventController.php]: :robot:

Comment (Naapperas) [reply app/Http/Controllers/EventController.php]: Do you have any suggestions? @toni-santos suggested `Inicia sessão para fazer esta ação` :musical_note:

File: app/Http/Controllers/QuestController.php (hunk 3 above)
Comment (Naapperas) [reply app/Http/Controllers/QuestController.php]: :robot:

**Comment by Naapperas** [comment]
> Actually, now that I think about it, are those error messages when the user is not logged in even required? The routs have the fortify middleware anyway

routs

**Comment by Naapperas** [comment]
> Actually, now that I think about it, are those error messages when the user is not logged in even required? The routs have the fortify middleware anyway

Perhaps these specific routes will never fall in that branch but better be safe than sorry.

**Review by CiscoPr — APPROVED** [review]

**Review by ttoino — APPROVED** [review]

**Pull request merged**